test/middleware/AuthMiddleware.test.ts

```diff
@@ -0,0 +1,127 @@
+
+const protectedPages = [
+    PENALTY_DETAILS_PAGE_URI,
+    OTHER_REASON_DISCLAIMER_PAGE_URI,
+    OTHER_REASON_PAGE_URI
+];
+
+describe('Authentication Middleware', () => {
+
+
+    describe('Authed path', () => {
+
+        let penaltyDetails: Record<string, any>;
+
+        before('sign in server should be running', async () => {
+
+            penaltyDetails = {
+                penaltyReference: 'A12345678',
+                companyNumber: 'SC123123'
+            };
+
+            await request(createApp())
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .catch(err => assert.fail(err));
+
+
+        });
+
+        it('should not redirect the user to the sign in page if the user is signed in', async () => {
+
+            for (const page of protectedPages) {
+                await request(authedApp).get(page)
+                    .expect(200);
+            }
+
+        });
+
+        it('should allow the user to go to the penalty reference screen if authed', async () => {
+
+            await request(authedApp)
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .expect(200);
+
+        });
+        it('should continue to the next page without requiring auth if penalty details are valid', async () => {
+
+            await request(authedApp)
+                .post(PENALTY_DETAILS_PAGE_URI)
+                .send(penaltyDetails)
+                .expect(_ => expect(_.header.location).to.include(OTHER_REASON_DISCLAIMER_PAGE_URI));
+
+        });
+    });
+    describe('Unauthed path', () => {
+
+        const appWithoutCookie = { label: 'App without Cookie', app: createApp(session, false) };
+        const unAuthedApp = { label: 'App without Cookie and Session', app: createApp() };
+
+        const unAuthedAppStateVariations = [appWithoutCookie, unAuthedApp];
+
+        before('sign in server should be running', async () => {
+
+            await request(createApp())
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .catch(err => assert.fail(err));
+        });
+
+        it('should redirect the user to sign in screen when trying to access protected pages', async () => {
+
+            for (const variation of unAuthedAppStateVariations) {
+                for (const page of protectedPages) {
+                    await request(variation.app).get(page)
+                        .expect(302).then(res => expect(res.header.location).to.include('/signin'));
+                }
+
+            }
+
+        });
+
+    });
+
+    describe('Auth Edge case', () => {
+
+        const appWithoutSessionObject = createApp(createFakeSession([], config.cookieSecret), true);
+
+        before('sign in server should be running', async () => {
+
+            await request(createApp())
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .catch(err => assert.fail(err));
+        });
+
+        it('should allow user to access protected pages if session object is somehow lost.', async () => {
```

Comment:
Not sure what you trying to do here. What did you mean by 'is somehow lost'? A `createApp(createFakeSession([], config.cookieSecret), true)` call prepares session as other tests.